src/main.rs

```diff
@@ -61,9 +64,12 @@ impl<'a, I> PixelColumn<'a, I> where I: GenericImage {
 	///
 	fn new(img: &'a I, column: u32) -> PixelColumn<'a, I> {
 		// column must be in range
-		assert!(column < img.width());
-
-		PixelColumn {column_index: column, img: img}
+        let w = img.width();
+        if column >= w {
+            Self::new(img, w - 1)
```

Comment:
It is programmers error to try to create column out of range just as it is error to index outside of some arr[]. Silently ignoring errors causes hard-to-find bugs. Therefore I used the same convention as arrays use (panic on out of bounds, not returning the last element).